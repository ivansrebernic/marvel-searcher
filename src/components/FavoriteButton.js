import React from 'react'
import styled from 'styled-components'
import FavIcon from '../assets/images/fav.png'
import FavIconToggled from '../assets/images/fav-toggled.png'


const FavoriteButtonStyled = styled.a`
    align-self:center;
    margin-left:1rem;
    margin-right: 1rem;
    &:hover{
        cursor: pointer;
    }
     img {
        width: 32px;
        height:32px;
    }
`




class FavoriteButton extends React.Component {


    constructor(props) {
        super(props)
        this.state = { isFavorite: false }
        this.toggleButton = this.toggleButton.bind(this);
    }


    toggleButton() {
        this.setState(state => ({
            isFavorite: !state.isFavorite
        }))
        console.log("asd")
    }

    render() {
        return (
            <FavoriteButtonStyled onClick={this.toggleButton} >
                <img src={this.state.isFavorite ? FavIconToggled : FavIcon} />
            </FavoriteButtonStyled >



        )
    }
}

export default FavoriteButton;