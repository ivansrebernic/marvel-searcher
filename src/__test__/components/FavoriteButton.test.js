import React from 'react'
import { shallow } from 'enzyme'
import FavoriteButton from '../../components/FavoriteButton'
import ProviderMock from '../../__mocks__/providerMock'
import { Simulate } from 'react-dom/test-utils';


describe('<FavoriteButton>', () => {
    it('Renders correctly', () => {
        const wrapper = shallow(<FavoriteButton></FavoriteButton>)
        expect(wrapper).toMatchSnapshot()
    });

})