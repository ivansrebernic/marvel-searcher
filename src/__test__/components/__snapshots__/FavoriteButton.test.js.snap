// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoriteButton> Renders correctly 1`] = `
<styled.a
  onClick={[Function]}
>
  <img
    alt="favorite button"
    src="fav.png"
  />
</styled.a>
`;
